nimbus_generator/example/lib/service/api_service.g.dart

```diff
@@ -0,0 +1,70 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+part of 'api_service.dart';
+
+// **************************************************************************
+// NimbusSupporterGenerator
+// **************************************************************************
+
+// ignore_for_file: unnecessary_brace_in_string_interps,no_leading_underscores_for_local_identifiers
+
+const _kServerName = 'ApiService';
+
+class _ApiService implements ApiService {
+  _ApiService({
+    required String baseUrl,
+    ApiServerHttpRequestBuilder? requestBuilder,
+    ApiServerHttpResponseParser? responseParser,
+  }) {
+    final urlSplit = baseUrl.split(':');
+    final host = urlSplit.first;
+    final port = int.tryParse(urlSplit.lastOrNull ?? '');
+    ApiRegistory.registApiServer(
+      ApiServerHttp(
+        name: _kServerName,
+        host: host,
+        port: port,
+        requestBuilder: requestBuilder,
+        responseParser: responseParser,
+      ),
+    );
+
+    ApiRegistory.registApi(
+      ApiHttp<Map<String, dynamic>, Map<String, dynamic>, Request, Response>(
+        name: 'login',
+        serverName: _kServerName,
+        method: HttpMethod.POST,
+        path: '/web/login.bf',
+        inputCreator: (_) => {},
+        outputCreator: (_) => {},
```

Comment:
In inputCreator and outputCreator, I assume that the API is used to create a DataSet association, but why is this empty and generated?

Reply:
I still use a DataSet association, but I don't want to develop must learn this concept so I hide that logic by using source_gen 

In the login function of the api_service.g.dart file, you can see two functions toJson and fromJson, that will auto-convert to Dataset and return the expected result. I implemented this inspired by https://pub.dev/packages/json_serializable

toJson
Model -> Dataset -> Map 
fromJson
Map -> Dataset -> Model

Reply:
The purpose is to increase code readability by declaring input DTO definitions and output DTO definitions in inputCreator and outputCreator in a declaration called API.
Therefore, an empty implementation of inputCreator and outputCreator is not the design intent of this library.